scripts/check_pre_commit_consistency.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# (C) Copyright Alpine Quantum Technologies GmbH 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# Check consistency between versions of the pre-commit hooks
+# and the packages installed by poetry.
+
+set -euo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```

Comment:
"readonly" would make this clearer. Plus points for bracket delimiting here though; would like to see them everywhere else too...